### PR TITLE
Improve error handling in server.js

### PR DESCRIPTION
**Pull request by gwicke**
- Handle exceptions throwing falsy values better
- Preserve full internal error objects & log them

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -189,19 +193,21 @@ function handleRequest (opts, req, resp) {
         return handleResponse(reqOpts, newReq, resp, result);
     })
     .catch (function(e) {
-        if (e && e.name !== 'HTTPError') {
-            var stack = e.stack;
+        if (!e || e.name !== 'HTTPError') {
```
Comment (d00rman) [inline lib/server.js]: This was a problem I have seen long ago but haven't gotten around to fix. Yey :)

File: lib/server.js [hunk 2]
```diff
@@ -122,6 +122,10 @@ function handleResponse (opts, req, resp, response) {
             req: req,
             msg: "No content returned"
         });
+
+        if (!response) { response = {}; }
+        if (!response.headers) { response.headers = {}; }
```
Comment (d00rman) [inline lib/server.js]: with these two lines in place, `response &&` can be removed from line 130.

Comment (gwicke) [inline lib/server.js]: Good catch, will amend.

Comment (hardikj) [inline lib/server.js]: Nicely spotted :+1:

**Comment by d00rman** [comment]
closing this in favour of #220

**Pull request closed without merging**